elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,1089 @@ function xml_form_elements_creative_commons($element, &$form_state) {
- *   'y' to allow 'n' to disallow.
+ *   Derivatives.
+ * @param string $commercial
+ *   Commercial.
  * @param string $jurisdiction
  *   Legal jurisdiction code.
+ * @param bool $override_version
+ *   TRUE to override version, FALSE to get newest.
+ * @param string $version
+ *   License version.
+ * @param string $license_uri
+ *   License URI.
  *
- * @return mixed
- *   SimpleXMLElement the return from the REST API.
- *   FALSE if failed request.
+ * @return array
+ *   The associative array representing the `creative_commons` input values.
  */
-function xml_form_elements_get_creative_commons($commercial, $derivatives, $jurisdiction, $disabled = FALSE) {
-  if ($disabled) {
-    return FALSE;
+function xml_form_elements_creative_commons_make_value_array($license, $derivatives, $commercial, $jurisdiction, $override_version, $version, $license_uri) {
+  return array(
+    'license_fieldset' => array(
+      'license_type' => $license,
+      'allow_derivatives' => $derivatives,
+      'allow_commercial' => $commercial,
+      'license_jurisdiction' => $jurisdiction,
+      'override_license_version' => $override_version,
+      'license_version' => $version,
+      'license_uri' => $license_uri,
+    ));
+}
+
+/**
+ * Get path for license.
+ *
+ * @param string $license
+ *   A string containing the license ('zero', 'mark', 'by' or 'sampling+').
+ *
+ * @return string|FALSE
+ *   The path where the license is found ('licenses' or 'publicdomain'),
+ *   or FALSE if $license was not recognized.
+ */
+function xml_form_elements_creative_commons_get_license_path($license) {
+  $publicdomain = array('zero', 'mark');
+  $licenses = array('by', 'sampling+');
+  if (in_array($license, $publicdomain)) {
+    return 'publicdomain';
+  }
+  elseif (in_array($license, $licenses)) {
+    return 'licenses';
+  }
+  return FALSE;
+}
+
+/**
+ * Test the Creative Commons version format validity.
+ *
+ * @param string $license_version
+ *   The license version.
+ *
+ * @return bool
+ *   TRUE if format is valid, FALSE otherwise.
+ */
+function xml_form_elements_creative_commons_license_version_has_valid_format($license_version) {
+  if (empty($license_version) || preg_match("#^[0-9]\.[0-9]$#", $license_version)) {
+    return TRUE;
+  }
+  return FALSE;
+}
+
+/**
+ * Test the Creative Commons license URI format validity.
+ *
+ * @param string $license_uri
+ *   The license URI.
+ *
+ * @return bool
+ *   TRUE if format is valid, FALSE otherwise.
+ */
+function xml_form_elements_creative_commons_license_uri_has_valid_format($license_uri) {
+  if (empty($license_uri) || (
+    preg_match("#^" . XML_FORM_ELEMENTS_CREATIVE_COMMONS_BASE_URL . "[a-z]+/[a-z+-]+/[0-9]\.[0-9]/([a-z]+/)?$#", $license_uri) &&
+    (!preg_match("#/by[-/]#", $license_uri) ||
+      preg_match("#^" . XML_FORM_ELEMENTS_CREATIVE_COMMONS_BASE_URL . xml_form_elements_creative_commons_get_license_path('by') . "/by((-nc)?(-nd|-sa?)|(-nd|-sa)?(-nc)?)/#", $license_uri))
```

Comment:
Sorry... Just realised that this line should read (switch `?` and `)` in columns 149-150):
```php
      preg_match("#^" . XML_FORM_ELEMENTS_CREATIVE_COMMONS_BASE_URL . xml_form_elements_creative_commons_get_license_path('by') . "/by((-nc)?(-nd|-sa)?|(-nd|-sa)?(-nc)?)/#", $license_uri))
```